lib/Runtime/Debug/TTRuntmeInfoTracker.cpp

```diff
@@ -419,13 +442,33 @@ namespace TTD
         }
     }
 
+    void ScriptContextTTD::ProcessFunctionBodyOnUnLoad(Js::FunctionBody* body, Js::FunctionBody* parent)
+    {
+        //if this is a root (parent is null) then put this in the rootbody pin set so it isn't reclaimed on us
+        if(parent == nullptr)
+        {
+            TTDAssert(this->m_ttdPinnedRootFunctionSet->Contains(body), "We already added this function!!!");
+            this->m_ttdPinnedRootFunctionSet->Remove(body);
```

Comment:
not clear why you're removing this after the assert here.  Should this be a fail-fast condition?

Reply:
TTDAssert is a fail-fast assert.